copy I915 header file directly

XinfengZhang opened the pull request.

dvrogozh committed:
i915: separate out vdbox balancing custom uAPI

This change moves vdbox balancing under MEDIA_EXT define. Configure
driver build as follows to get support of this feature:
  cmake -DMEDIA_EXT=ON /path/to/media-driver
By default it is disabled

Change-Id: Ia89e4beb1d3bdf681efbf40d9fb5159fe3663be3
Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>

dvrogozh committed:
i915: separate out sseu programming custom uAPI

This change moves sseu under MEDIA_EXT define. Configure
driver build as follows to get support of this feature:
  cmake -DMEDIA_EXT=ON /path/to/media-driver
By default it is disabled.

Change-Id: Ifdba7d12207a075fad11722fce4e7bc547b9b227
Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>

dvrogozh committed:
i915: cleanup mock headers - remove Linux customizations

Change-Id: Ic9d70835e3c51a79b1f0404e13f861d920c8f7ea
Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>

dvrogozh committed:
i915: cleanup mock - remove Android customizations

This patch removes all Android customizations except pad_to_size
support which actually landed upstream. It will be enabled in one
of the next patches.

Change-Id: Ib7c8b190e35ee839c95de21863e4ff58a827d75e
Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>

dvrogozh committed:
i915: remove custom unused DRM_I915_PERFMON ioctl

i915 driver got DRM_I915_PERF_* ioctls support. Custom  DRM_I915_PERFMON
should no longer be needed.

Change-Id: I2c1538c84b4565eacfcc94ee5c85c8c267f393f2
Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>

dvrogozh committed:
i915: remove unused custom Android ioctls

These ioctls seems not being used anywhere in media driver:
 * DRM_I915_GEM_ACCESS_USERDATA
 * define DRM_I915_CMD_PARSER_APPEND
 * define DRM_I915_EXT_IOCTL

Change-Id: I729404ac854fe6510c719a1fd0dfef577800d960
Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>

dvrogozh committed:
i915: update EXEC_OBJECT_PAD_TO_SIZE support from upstream

Upstream i915 driver got support for pad_to_size.

WARNING: this patch changes definition of EXEC_OBJECT_PAD_TO_SIZE
from (custom):
  #define EXEC_OBJECT_PAD_TO_SIZE (1<<4)
to (upstream):
  #define EXEC_OBJECT_PAD_TO_SIZE (1<<5)
It is required for Android kernels to be adjusted correspondingly, i.e.
aligned with the afficial upstream version.

Change-Id: Ibf4c06b792cda2d4a117aa46411c0db9f6afdbfd
Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>

dvrogozh committed:
i915: sync headers with upstream

After this change:
 * i915_drm_mock.h - fully synced with upstream
 * i915_drm.h - still contains few Android customizations

Change-Id: I6fba9ce1691b362128ea5deedaf63737f69ed9bb
Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>

dvrogozh commented on media_driver/linux/common/os/libdrm/include/i915_header.h:
```diff
@@ -0,0 +1,32 @@
+#ifndef _I915_HEADER_H_
+#define _I915_HEADER_H_
+#ifndef __user
+#define __user
```
What's the purpose of the __user? And what's the purpose of i915_header.h?

XinfengZhang replied:
we want the i915_drm.h is bitmatch with upstreamed drm_tip. but in the header , there are key word "__user" right? , instead of removing it, I prefer to add a define, it will delete it automatically.

dvrogozh reviewed (changes requested):
I believe there were other places in i915 API implementation which we would like to cleanup (Android customizations).

XinfengZhang commented:
hi @dvrogozh , long term target is remove all #ifdef ANDROID in whole driver (DDI, HAL, OS etc.)
short term target, remove it from i915_drm.h.

XinfengZhang committed:
i915: sync i915_drm.h header with upstream

Signed-off-by: XinfengZhang <carl.zhang@intel.com>

XinfengZhang committed:
copy i915_drm.h directly from drm_tip

i915_drm.h is from drm_tip/include/uapi/drm/i915_drm.h
commit id is cd956bfcd0f58d20485ac0a785415f7d9327a95f

Signed-off-by: XinfengZhang <carl.zhang@intel.com>

XinfengZhang committed:
[Decode]ULT remove libdrm_mock header files

directly refer the header file in mos folder

Signed-off-by: XinfengZhang <carl.zhang@intel.com>

dvrogozh commented on media_driver/linux/common/os/libdrm/include/i915_drm_ext.h:
```diff
@@ -61,6 +61,41 @@ struct drm_i915_gem_context_param_sseu {
+    __u32 pad;
+    __u32 flags;
+#define I915_CREATE_PLACEMENT_NORMAL    0 /* standard swappable bo  */
+#define I915_CREATE_PLACEMENT_STOLEN    1 /* Cannot use CPU mmaps */
```
I thought you said that STOLEN is not used at the moment? Does that mean we can entirely remove `__flags` right now?

I believe that answer is probably "Yes". Here is some grep results:
```
[dvrogozh@dvrscl media-driver]$ fgrep -rsn I915_CREATE_PLACEMENT_NORMAL media_driver/
media_driver/linux/common/os/libdrm/include/i915_drm.h:654:#define I915_CREATE_PLACEMENT_NORMAL    0 /* standard swappable bo  */
[dvrogozh@dvrscl media-driver]$ fgrep -rsn I915_CREATE_PLACEMENT_STOLEN  media_driver/
media_driver/linux/common/os/libdrm/include/i915_drm.h:655:#define I915_CREATE_PLACEMENT_STOLEN    1 /* Cannot use CPU mmaps */
media_driver/linux/common/os/libdrm/mos_bufmgr.c:1012:            create.flags |= I915_CREATE_PLACEMENT_STOLEN;
[dvrogozh@dvrscl media-driver]$ fgrep -rsn I915_CREATE_PLACEMENT_MASK  media_driver/
media_driver/linux/common/os/libdrm/include/i915_drm.h:656:#define I915_CREATE_PLACEMENT_MASK      0xff
[dvrogozh@dvrscl media-driver]$ fgrep -rsn I915_CREATE_POPULATE  media_driver/
media_driver/linux/common/os/libdrm/include/i915_drm.h:657:#define I915_CREATE_POPULATE            (1<<8) /* Pre-populate object pages */
media_driver/linux/common/os/libdrm/mos_bufmgr.c:1015:            create.flags |= I915_CREATE_POPULATE;
media_driver/linux/common/os/libdrm/mos_bufmgr.c:1040:                create.flags |= I915_CREATE_POPULATE;
[dvrogozh@dvrscl media-driver]$ fgrep -rsn I915_CREATE_FLUSH  media_driver/
media_driver/linux/common/os/libdrm/include/i915_drm.h:658:#define I915_CREATE_FLUSH               (1<<9) /* Clflush prepopulated pages */
media_driver/linux/common/os/libdrm/include/i915_drm.h:659:#define __I915_CREATE_UNKNOWN_FLAGS     -(I915_CREATE_FLUSH << 1)
media_driver/linux/common/os/libdrm/mos_bufmgr.c:1018:            create.flags |= I915_CREATE_FLUSH;
media_driver/linux/common/os/libdrm/mos_bufmgr.c:1043:                create.flags |= I915_CREATE_FLUSH;
[dvrogozh@dvrscl media-driver]$ fgrep -rsn BO_ALLOC_STOLEN  media_driver/
media_driver/linux/common/ddi/media_libva_util.cpp:532:            bo = mos_bo_alloc_tiled(mediaDrvCtx->pDrmBufMgr, "MEDIA", gmmPitch, gmmSize/gmmPitch, 1, &tileformat, (unsigned long *)&ulPitch, BO_ALLOC_STOLEN);
media_driver/linux/common/os/libdrm/include/mos_bufmgr.h:133:#define BO_ALLOC_STOLEN        (1<<1)
media_driver/linux/common/os/libdrm/mos_bufmgr.c:916:    if (flags & BO_ALLOC_STOLEN) {
media_driver/linux/common/os/libdrm/mos_bufmgr.c:1011:        if (flags & BO_ALLOC_STOLEN)
media_driver/linux/common/os/libdrm/mos_bufmgr.c:1025:            if (flags & BO_ALLOC_STOLEN) {
media_driver/linux/common/os/libdrm/mos_bufmgr.c:1083:    if (flags & BO_ALLOC_STOLEN)
media_driver/linux/common/os/mos_graphicsresource_specific.cpp:281:                 &tileFormatLinux, &linuxPitch, BO_ALLOC_STOLEN);
media_driver/linux/common/os/mos_os_specific.c:2087:                 &tileformat_linux, &ulPitch, BO_ALLOC_STOLEN);
[dvrogozh@dvrscl media-driver]$ fgrep -rsn BO_ALLOC_POPULATE  media_driver/
media_driver/linux/common/os/libdrm/include/mos_bufmgr.h:134:#define BO_ALLOC_POPULATE   (1<<2)
media_driver/linux/common/os/libdrm/mos_bufmgr.c:1014:        if (flags & BO_ALLOC_POPULATE)
media_driver/linux/common/os/libdrm/mos_bufmgr.c:1039:            if (flags & BO_ALLOC_POPULATE)
[dvrogozh@dvrscl media-driver]$ fgrep -rsn BO_ALLOC_FLUSH  media_driver/
media_driver/linux/common/os/libdrm/include/mos_bufmgr.h:135:#define BO_ALLOC_FLUSH        (1<<3)
media_driver/linux/common/os/libdrm/mos_bufmgr.c:994:    if (alloc_from_cache && (flags & BO_ALLOC_FLUSH))
media_driver/linux/common/os/libdrm/mos_bufmgr.c:1017:        if (flags & BO_ALLOC_FLUSH)
media_driver/linux/common/os/libdrm/mos_bufmgr.c:1042:            if (flags & BO_ALLOC_FLUSH)
```
So, the only usage is for BO_ALLOC_STOLEN, but if you will take a look to the code:
https://github.com/dvrogozh/media-driver/blob/e18f84c8fd866a2f789da2b2f68515d2c20987e0/media_driver/linux/common/os/mos_os_specific.c#L1810
and
https://github.com/dvrogozh/media-driver/blob/e18f84c8fd866a2f789da2b2f68515d2c20987e0/media_driver/linux/common/os/mos_graphicsresource_specific.cpp#L262

you will see that it is covered by I915_PARAM_CREATE_VERSION and I don't see that it is defined anywhere in the driver:
```
[dvrogozh@dvrscl media-driver]$ fgrep -rsn I915_PARAM_CREATE_VERSION
media_driver/linux/common/ddi/media_libva_util.cpp:522:#if defined(I915_PARAM_CREATE_VERSION)
media_driver/linux/common/ddi/media_libva_util.cpp:528:        gp.param = I915_PARAM_CREATE_VERSION;
media_driver/linux/common/os/mos_graphicsresource_specific.cpp:266:#if defined(I915_PARAM_CREATE_VERSION)
media_driver/linux/common/os/mos_graphicsresource_specific.cpp:274:    gp.param = I915_PARAM_CREATE_VERSION;
media_driver/linux/common/os/mos_os_specific.c:2073:#if defined(I915_PARAM_CREATE_VERSION)
media_driver/linux/common/os/mos_os_specific.c:2081:    gp.param = I915_PARAM_CREATE_VERSION;
```

dvrogozh replied:
So, I suggest to attempt to simply remove __flags and related code. How do you think?
The problem here is that mos_gem_bo_bucket_for_size() implementations diverged between Linux/Android. I am not sure how safe we would be to reuse Linux variant on Android. We should, but...
